gabbi/driver.py

```diff
@@ -151,15 +152,43 @@ def py_test_generator(test_dir, host=None, port=8001, intercept=None,
                         prefix=prefix, require_ssl=require_ssl,
                         url=url)
 
+    test_list = []
     for test in tests:
         if hasattr(test, '_tests'):
             # Establish fixtures as if they were tests. These will
             # be cleaned up by the pytester plugin.
-            yield 'start_%s' % test._tests[0].__class__.__name__, \
-                test.start, result
+            test_list.append(('start_%s' % test._tests[0].__class__.__name__,
+                              test.start, result))
             for subtest in test:
-                yield '%s' % subtest.__class__.__name__, subtest, result
-            yield 'stop_%s' % test._tests[0].__class__.__name__, test.stop
+                test_list.append(('%s' % subtest.__class__.__name__,
+                                  subtest, result))
+            test_list.append(('stop_%s' % test._tests[0].__class__.__name__,
+                              test.stop))
+
+    if metafunc:
+        if metafunc.function == test_pytest:
```

Comment:
Then other tests which may be in the same file carry on like normal tests without changes.

I tried to get rid of the conditional and just do a "modify everything!" style to see about not having `test_pytest`. The code in this block is only run if there is at least one `test_*` in the modules scope.